### ISSUE TITLE
./dspectrum:66:in `block in listen': undefined method `chomp' for nil:NilClass (NoMethodError)

### ISSUE THREAD

**Issue by cn0xroot** [tresacton/dspectrum#1]
nice to talk to you , dspectrum is so cool! but when i used it:

```
./dspectrum
dspectrum - by nullwolf
USAGE: 
    help: Prints this message
    c: Continue processing signals from Inspectrum
    inspect: Compare the next two signals from currently opened file
    dual: Compare next signal from this file with a signal from another file (re-spawns Inspectrum in between)
    stop: Closes Inspectrum
    start: Opens Inspectrum
    verbose: Include the array of raw data from Inspectrum
    exit: Quit dspectrum
[Enter command, or type 'help'] >  start
Spawning Inspectrum
[Enter command, or type 'help'] >  inspect
./dspectrum:66:in `block in listen': undefined method `chomp' for nil:NilClass (NoMethodError)
    from ./dspectrum:63:in `loop'
    from ./dspectrum:63:in `listen'
    from ./dspectrum:224:in `single_file_inspect'
    from ./dspectrum:208:in `control'
    from ./dspectrum:19:in `block in start'
    from ./dspectrum:17:in `spawn'
    from ./dspectrum:17:in `start'
    from ./dspectrum:204:in `control'
    from ./dspectrum:19:in `block in start'
    from ./dspectrum:17:in `spawn'
    from ./dspectrum:17:in `start'
    from ./dspectrum:262:in `<main>'
```

how to fix it?Looking forward to your reply :)

**Comment by tresacton** [comment]
Thanks :)

May I ask which version of Inspectrum you are using, and what operating
system you have? I can't reproduce the error with the version I have
installed.

Once I can reproduce it, I'll be able to fix it.

Just as a side note, you don't need to enter the command 'start' unless
you've already closed the instance of Inspectrum that spawns when the
script starts up, or if you've previously issued the 'stop' command. All
'start' does is open a fresh instance of Inspectrum to monitor.

**Comment by cn0xroot** [comment]
the bug has fxed when i rebuild Inspectrum again.
but with a new question now:

```
 c

---------------------------------------------
[*] Processed At: 2016-09-23 12:21:07 +0800
---------------------------------------------


[-] Safety checking threshold:  0.014902368421052631
[!] Measurement for a 1 is in between the threshold and the upper safety limit. Occured: 43 times
[!] The issues listed above point to a low confidence level that I was able to guess the correct threshold. You'll want to manually verify these results.


[*] Binary:         1111111111111111111111111111111111111111111111111111111111111111111111111111
[*] Hexcode:        fffffffffffffffffff
[*] Formatted Hexcode:  \xff\xff\xff\xff\xff\xff\xff\xff\xff\xf
[*] Ascii:          ?????????f
```

![](http://image.3001.net/images/20160923/14746046023149.png)

**Comment by cn0xroot** [comment]
Screenshot： http://image.3001.net/images/20160923/14746046023149.png

Two results are not the same
https://camo.githubusercontent.com/1a7d46fa3851a486245f045e1861416615b85587/687474703a2f2f696d6167652e333030312e6e65742f696d616765732f32303136303930372f31343733323238333638393834362e706e67

**Comment by tresacton** [comment]
I think you're going to be my favourite new beta tester xD 

Can you please give the attached a try, and let me know if it works / you get the same result? I've included a screenshot so you can see how many symbols I've lined up in the cursors grid.

[_] Binary:         11001101000011
[_] Hexcode:        3343
[_] Formatted Hexcode:  \x33\x43
[_] Ascii:          3C

[ShortTest.zip](https://github.com/tresacton/dspectrum/files/488868/ShortTest.zip)

**Comment by cn0xroot** [comment]
it work http://image.3001.net/images/20160923/1474607099621.png

thanks!

**Comment by tresacton** [comment]
That's good to know! Thanks for the response. If you'd like me to further investigate the issue you're having with that file, please feel free to pop it up on pastebin or something and send me a link. In the mean time, I'll close this thread off =)

**Comment by cn0xroot** [comment]
my doorbell signal file has upload to https://pan.baidu.com/s/1kVBBW6V you can click"下载" to download it.

when set the symbols value =14 it work fine http://image.3001.net/images/20160923/1474607747229.png

but set the symbols value =76 

```
[Enter command, or type 'help'] >  c

---------------------------------------------
[*] Processed At: 2016-09-23 13:16:44 +0800
---------------------------------------------


[-] Safety checking threshold:  0.012964855263157896
[!] Measurement for a 1 is in between the threshold and the upper safety limit. Occured: 43 times
[!] The issues listed above point to a low confidence level that I was able to guess the correct threshold. You'll want to manually verify these results.


[*] Binary:         1111111111111111111111111111111111111111111111111111111111111111111111111111
[*] Hexcode:        fffffffffffffffffff
[*] Formatted Hexcode:  \xff\xff\xff\xff\xff\xff\xff\xff\xff\xf
[*] Ascii:          ?????????f

```

http://image.3001.net/images/20160923/14746078424047.png

maybe this is a bug.

**Comment by tresacton** [comment]
Very strange indeed. 

I gave this a shot in my environment, and it seemed to work out perfectly. I manually decoded the bits by sight and got the same result as the tool.

I suspect the reason you are having difficulty may be that you haven't quite aligned the cursors with the start and the end of your bits. It needs to be precise for Inspectrum to appropriately output the amplitude values. I have included an example of how I aligned the bits in the attached screenshot, in addition to the output. I hope this helps

<img width="839" alt="screen shot 2016-09-23 at 3 55 27 pm" src="https://cloud.githubusercontent.com/assets/551449/18775638/5f56c308-81a6-11e6-9c44-21c07d582c15.png">
<img width="1440" alt="screen shot 2016-09-23 at 3 55 18 pm" src="https://cloud.githubusercontent.com/assets/551449/18775639/5f5ca854-81a6-11e6-8876-a57c2600cdb5.png">
